cmd/root.go

```diff
@@ -176,10 +181,16 @@ func startServer(opts *common.Options) error {
 	var blockHeight int
 	var chainName string
 	var branchID string
+	var rpcClient *rpcclient.Client
+	var err error
 	if opts.Darkside {
 		chainName = "darkside"
 	} else {
-		rpcClient, err := frontend.NewZRPCFromConf(opts.ZcashConfPath)
+		if opts.RPCUser != "" && opts.RPCPassword != "" && opts.RPCHost != "" && opts.RPCPort != "" {
+			rpcClient, err = frontend.NewZRPCFromFlags(opts)
```

Comment:
If the user specified 3 of these but forgot one, then we'd get the values from the conf file, but would that be confusing to the user? Maybe throw an error if any of these 4 are specified on the command line but not all of them. 

OR, a really nice thing would be to take the values from the file (if it's specified, I understand that the goal of this PR is that the conf file doesn't have to be specified, doesn't have to exist on this machine), and then the command-line options _override_ the values read from the file. That's how `zcashd` works -- command-line arguments override config file values.

But if you think that's too much, that could be a separate PR.

One really minor suggestion is to refactor this test (if all the strings are non-empty) into a separate function that can be called from two places (here and at line 69).

Reply:
Agree with the comments, opened https://github.com/zcash/lightwalletd/issues/291 to address it.